comid-triples.cddl

```diff
@@ -0,0 +1,156 @@
+concise-mid-tag = {
+  ; TODO(tho) language
+  ; TODO(tho) tag-id
+  ; TODO(tho) module-name
+  ; TODO(tho) entity
+  ; TODO(tho) linked-tags
+  comid.triples => triples-map
+  ; TODO(tho) extension
+}
+
+triples-map = non-empty<{
+  ? comid.reference-triples => one-or-more<reference-triple-record>
+  ? comid.endorsed-triples => one-or-more<endorsed-triple-record>
+  ? comid.identity-triples => one-or-more<identity-triple-record>
+  * $$triples-map-extension
+}>
+
+; REFERENCE-VALUE triple
+; "target environment $TE" "has reference measurements" "$RV"
+reference-triple-record = [
+  environment-map ; target environment
+  one-or-more<measurement-map> ; reference measurements
+]
+
+; ENDORSED-VALUE triple
+; "environment $E" "has endorsed measurements" "$EV"
+endorsed-triple-record = [
+  environment-map ; (target or attesting) environment
+  one-or-more<measurement-map> ; endorsed measurements
+]
+
+; DEVICE-IDENTITY triple
+; (Should it be AE-IDENTITY instead?)
+; "attesting environment $AE" "signs Evidence with key" "$K"
+identity-triple-record = [
+  environment-map ; attesting-environment
+  one-or-more<COSE_Key> ; identity
+]
+
+environment-map = non-empty<{
+  ? comid.class => class-map
+  ? comid.instance => instance-map
+}>
+
+class-map = {
+  comid.class-id => $class-id-type-choice
+  ? comid.layer => uint
+  ? comid.index => uint
```

Comment:
as discussed: add optional vendor (tstr)

Reply:
as discussed: add optional model (tstr)

Reply:
We had past discussions that the attributes used to name an environment class shouldn't be extensible as that would create a bit of confusion for verifiers if there are multiple optional ways to name things. Combinatorics explosion. X.509 RDN and subAltName are a bit like this. We said we would use the extension point to include Dice specific fields. But that opens the door for other things.